drivers/reset/reset_npm1300.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_npm1300_reset
+
+#include <errno.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/mfd/npm1300.h>
+
+/* nPM1300 MAIN base address */
+#define NPM_MAIN_BASE 0x00U
+
+/* nPM1300 GPIO registers offsets */
+#define NPM_MAIN_OFFSET_SWRESET 0x01U
+
+struct reset_npm1300_config {
+	const struct device *mfd;
+};
+
+static int reset_npm1300_toggle(const struct device *dev, uint32_t id)
```

Comment:
`ARG_UNUSED(id)`?

I wonder if the reset API is well suited for this device (not sure what reset does in this case). The Reset API is oriented to IP blocks that control the reset lines of certain peripherals. See for example the DT-based API, where you'll typically have reset clients via `resets = <&resetctl ID>`. You can find example usages in ST/GD32 drivers.

Reply:
In this case it forces a full reset of the PMIC.  I agree it is tricky to decide where to put it.

Reply:
maybe as an API in the mfd driver?

Reply:
How about adding reset & hibernate to the mfd driver (as they are fairly specific to the npm1300), and adding ship to regulators (as it is common to many devices)?

Reply:
sounds good to me